### ISSUE TITLE
swarm fails to start 

### ISSUE THREAD

**Issue by jojimt** [contiv/ansible#138]
Seeing this issue consistently for the past couple of weeks:
Mar 22 08:42:09 openstack-01.muc.cisco.com swarm.sh[10755]: Error: failed to stop containers: [swarm-manager]
Mar 22 08:42:10 openstack-01.muc.cisco.com swarm.sh[10755]: Error response from daemon: no such id: swarm-manager
Mar 22 08:42:10 openstack-01.muc.cisco.com swarm.sh[10755]: Error: failed to remove containers: [swarm-manager]
Mar 22 08:42:10 openstack-01.muc.cisco.com swarm.sh[10755]: Error response from daemon: no such id: swarm-agent
Mar 22 08:42:10 openstack-01.muc.cisco.com swarm.sh[10755]: Error: failed to stop containers: [swarm-agent]
Mar 22 08:42:10 openstack-01.muc.cisco.com swarm.sh[10755]: Error response from daemon: no such id: swarm-agent
Mar 22 08:42:10 openstack-01.muc.cisco.com swarm.sh[10755]: Error: failed to remove containers: [swarm-agent]
Mar 22 08:42:10 openstack-01.muc.cisco.com systemd[1]: swarm.service: control process exited, code=exited status=1
Mar 22 08:42:10 openstack-01.muc.cisco.com systemd[1]: Unit swarm.service entered failed state.
Mar 22 08:42:10 openstack-01.muc.cisco.com systemd[1]: swarm.service failed.

rm -rf /var/lib/docker and reinstall fixes the issue.

**Comment by mapuri** [comment]
@jojimt thanks for reporting the issue. 

Can you please paste the complete service logs? These seems for for stopping the service, we will need the logs for service start.

**Comment by erikh** [comment]
@jojimt not sure if this is related, but wanted to make a note of it; the containers on hosts with the tests being run on them will lose swarm because the they tear down all the containers...

**Comment by jojimt** [comment]
@mapuri here is the complete log from another instance:

Mar 22 10:41:01 netmaster swarm.sh[3499]: starting swarm as master on node1[172.31.154.208]
Mar 22 10:41:01 netmaster swarm.sh[3499]: Error response from daemon: Could not find container for entity id 1f36313817055445447ad84f1f9ccc11173256db68dea65b0578a9b720217803
Mar 22 10:41:01 netmaster swarm.sh[3499]: Error response from daemon: Could not find container for entity id b07e61250ce71b382e1296b29c64c577a60118e539bcc7337e1baf24ec6be626
Mar 22 10:41:01 netmaster systemd[1]: swarm.service: main process exited, code=exited, status=1/FAILURE
Mar 22 10:41:01 netmaster swarm.sh[3518]: Error response from daemon: no such id: swarm-manager
Mar 22 10:41:01 netmaster swarm.sh[3518]: Error: failed to stop containers: [swarm-manager]
Mar 22 10:41:01 netmaster swarm.sh[3518]: Error response from daemon: no such id: swarm-manager
Mar 22 10:41:01 netmaster swarm.sh[3518]: Error: failed to remove containers: [swarm-manager]

@erikh this is in customer env with bare metal servers (no test scripts involved). @shaleman helped debug/workaround. Only way out is rm /var/lib/docker.

**Comment by jojimt** [comment]
Here is another one:
Mar 07 03:40:22 contiv-aci-server4.insieme.local systemd[1]: Started Swarm.
Mar 07 03:40:22 contiv-aci-server4.insieme.local systemd[1]: Starting Swarm...
Mar 07 03:40:22 contiv-aci-server4.insieme.local swarm.sh[21972]: starting swarm as master on node2[172.31.153.49]
Mar 07 03:40:22 contiv-aci-server4.insieme.local swarm.sh[21972]: Error response from daemon: Could not find container for entity id 9c8fc964fa14d475f613320e0f55f54ab7b6936a78d01e05a9e596801d00be6d
Mar 07 03:40:22 contiv-aci-server4.insieme.local swarm.sh[21972]: Error response from daemon: Could not find container for entity id f33bfb6edab67328836936ae50c2d037de6ac6b60d196f3db873d4ea103c2f24
Mar 07 03:40:22 contiv-aci-server4.insieme.local systemd[1]: swarm.service: main process exited, code=exited, status=1/FAILURE
Mar 07 03:40:22 contiv-aci-server4.insieme.local swarm.sh[21984]: Error response from daemon: no such id: swarm-manager
Mar 07 03:40:22 contiv-aci-server4.insieme.local swarm.sh[21984]: Error: failed to stop containers: [swarm-manager]

**Comment by mapuri** [comment]
@jojimt 
is this with `net_demo_installer` or even ansible/master in general is failing? Also is this only with Ubuntu?

The error seems to indicate that docker is failing the request. Can we get docker daemon logs as well around that time?

also this docker issue seems related: https://github.com/docker/docker/issues/17691

**Comment by jojimt** [comment]
People are using this with net_demo_installer which invokes ansible. So far, I have seen this with centos -- haven't used/seen people use ubuntu recently. This issue started ever since we started running swarm as container. We need to fix this as everyone is encountering it now.

**Comment by mapuri** [comment]
sure, can you please provide docker logs so we can debug this?

**Comment by jojimt** [comment]
you can look at contiv-aci-server4 (look for log entries on Mar 07)

**Comment by mapuri** [comment]
thanks, from docker logs it looks like docker was using overlayfs around that time, so still seems  related docker/docker#17691 which is fixed in 1.10

net_demo_installer will need to set `docker_version: 1.10.3` to move to new docker version. Can you please try? 

```
Mar 07 03:40:15 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:15.537024836-08:00" level=info msg="Daemon has completed initialization"
Mar 07 03:40:15 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:15.537066221-08:00" level=info msg="Docker daemon" commit=a34a1d5 execdriver=native-0.2 graphdriver=ov
Mar 07 03:40:15 contiv-aci-server4.insieme.local systemd[1]: Started Docker Application Container Engine.
-- Subject: Unit docker.service has finished start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit docker.service has finished starting up.
--
-- The start-up result is done.
Mar 07 03:40:19 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:19.465478688-08:00" level=info msg="POST /v1.21/images/create?fromImage=swarm%3A1.1.2"
Mar 07 03:40:22 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:22.638931398-08:00" level=info msg="POST /v1.21/containers/create?name=swarm-manager"
Mar 07 03:40:22 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:22.642404634-08:00" level=error msg="Handler for POST /v1.21/containers/create returned error: Could n
Mar 07 03:40:22 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:22.642448596-08:00" level=error msg="HTTP Error" err="Could not find container for entity id 9c8fc964f
Mar 07 03:40:22 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:22.689320719-08:00" level=info msg="POST /v1.21/containers/create?name=swarm-agent"
Mar 07 03:40:22 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:22.691821539-08:00" level=error msg="Handler for POST /v1.21/containers/create returned error: Could n
Mar 07 03:40:22 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:22.691861693-08:00" level=error msg="HTTP Error" err="Could not find container for entity id f33bfb6ed
Mar 07 03:40:22 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:22.767397590-08:00" level=info msg="POST /v1.21/containers/swarm-manager/stop?t=10"
Mar 07 03:40:22 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:22.767677465-08:00" level=error msg="Handler for POST /v1.21/containers/swarm-manager/stop returned er
Mar 07 03:40:22 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:22.767721638-08:00" level=error msg="HTTP Error" err="no such id: swarm-manager" statusCode=404
Mar 07 03:40:22 contiv-aci-server4.insieme.local docker[21604]: time="2016-03-07T03:40:22.836818868-08:00" level=info msg="DELETE /v1.21/containers/swarm-manager"

```

**Comment by mapuri** [comment]
just verified in a setup that setting `decoker_version: 1.10.3` fixes the issue. We need to set this version in net_demo_installer for Centos setups. Will open an issue there and close this one.
